heartbeat/tests/system/test_monitor.py

```diff
@@ -41,34 +42,39 @@ def test_http(self, status_code):
         self.assert_fields_are_documented(output[0])
 
     @parameterized.expand([
-        ("8185", "up"),
-        ("8186", "down"),
+        (lambda server: "localhost:{}".format(server.server_port), "up"),
+        # This IP is reserved in IPv4
+        (lambda server: "207.254.52.220:1233", "down"),
     ])
-    def test_tcp(self, port, status):
+    def test_tcp(self, url, status):
         """
         Test tcp server
         """
         server = self.start_server("hello world", 200)
-        self.render_config_template(
-            monitors=[{
-                "type": "tcp",
-                "hosts": ["localhost:" + port],
-            }],
-        )
+        try:
+            self.render_config_template(
+                monitors=[{
+                    "type": "tcp",
+                    "hosts": [url(server)],
+                    "timeout": "3s"
+                }],
+            )
 
-        proc = self.start_beat()
-        self.wait_until(lambda: self.log_contains("heartbeat is running"))
+            proc = self.start_beat()
+            try:
```

Comment:
Would this try be needed also in test_reload?

Reply:
Yes, I'm planning on going back and refactoring all the tests along those lines, but I didn't want to blow this PR up. In general the heartbeat python tests don't use `try/finally` well.